ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -363,14 +364,16 @@ private module Cached {
       or
       c.getAnArgument() instanceof CfgNodes::ExprNodes::PairCfgNode
     } or
-    TSynthSplatArgumentNode(CfgNodes::ExprNodes::CallCfgNode c) {
-      exists(Argument arg, ArgumentPosition pos | pos.isPositional(_) | arg.isArgumentOf(c, pos)) and
-      not exists(Argument arg, ArgumentPosition pos | pos.isSplat(_) | arg.isArgumentOf(c, pos))
+    TSynthSplatArgumentNode(CfgNodes::ExprNodes::CallCfgNode c) or
+    TSynthSplatArgumentElementNode(CfgNodes::ExprNodes::CallCfgNode c, int n) {
+      n in [0 .. 10] and
+      exists(Argument arg, ArgumentPosition pos | pos.isSplat(_) and arg.isArgumentOf(c, pos))
```

Comment:
I believe this should be limited to `pos.isSplat(any(int splatPos | splatPos > 0))`?

Reply:
Yeah, I was thinking we would still have `ppos.isSynthSplat() and apos.isSplat(0)`, since there should be no need to go through a `TSynthSplatArgumentElementNode` node in that case, as the indices don't need to be adjusted.